nutils/solver.py

```diff
@@ -394,8 +388,12 @@ class newton(cache.Recursion, length=1):
         (boolean) or NaN (float). In the remaining positions the values of
         ``lhs0`` are returned unchanged (boolean) or overruled by the values in
         `constrain` (float).
-    linesearch : :class:`nutils.solver.LineSearch`
-        Callable that defines relaxation logic.
+    linesearch : Callable[(float, float, float, float), float]
```

Comment:
I think the return value is incorrect. Shouldn't this be `Tuple[float, bool]`? Also, in the [docs for `Callable`](https://docs.python.org/3/library/typing.html#typing.Callable) a list is used as first argument, not a tuple. The docs don't really state this as a requirement, though.

Reply:
fixed both issues

Reply:
The return type is now `bool`, not `Tuple[float, bool]`.

Reply:
fix for real this time